tests/adapter/dbt/tests/adapter/materialized_views/test_materialized_views.py

```diff
@@ -74,3 +81,54 @@ def test_updated_base_table_data_only_shows_in_materialized_view_after_rerun(sel
             self.get_records(project, self.materialized_view)
             == self.starting_records + self.inserted_records
         )
+
+
+class MaterializedViewTestsSkipConfigChangeBase(MaterializedViewTestsBase):
+    @pytest.fixture(scope="class")
+    def project_config_update(self):
+        return {"models": {"on_configuration_change": "skip"}}
+
+    def test_on_configuration_change_skips_with_update(self, project):
+        results = run_dbt(
+            ["run", "--models", self.materialized_view, "--vars", "quoting: {identifier: True}"]
+        )
+        assert len(results.results) == 1
+        result = results.results[0]
+        assert result.node.config.on_configuration_change == "skip"
+        assert result.status == RunStatus.Success
+        assert result.adapter_response["rows_affected"] == -1
+
+
+class TestMaterializedViewSkipTestsBase(MaterializedViewTestsSkipConfigChangeBase):
+    @pytest.mark.skip("This currently fails since we're mocking with a traditional view")
+    def test_updated_base_table_data_only_shows_in_materialized_view_after_rerun(self, project):
+        pass
+
+    def test_on_configuration_change_skips_with_update(self, project):
+        super().test_on_configuration_change_skips_with_update(project)
+
+    def test_relation_is_materialized_view_when_rerun(self, project):
+        super().test_relation_is_materialized_view_when_rerun(project)
+
+
+class MaterializedViewTestsFailConfigChangeBase(MaterializedViewTestsBase):
+    @pytest.fixture(scope="class")
+    def project_config_update(self):
+        return {"models": {"on_configuration_change": "fail"}}
+
+    def test_on_configuration_change_fails_with_update(self, project):
+        results = run_dbt(
+            ["run", "--models", self.materialized_view, "--vars", "quoting: {identifier: True}"],
+            expect_pass=False,
+        )
+        results.results[0].node.config.on_configuration_change == "fail"
+        assert results.results[0].status == RunStatus.Error
+
+
+class TestMaterializedViewFailTestsBase(MaterializedViewTestsFailConfigChangeBase):
+    @pytest.mark.skip("This currently fails since we're mocking with a traditional view")
+    def test_updated_base_table_data_only_shows_in_materialized_view_after_rerun(self, project):
+        pass
+
+    def test_on_configuration_change_fails_with_update(self, project):
```

Comment:
We can make this last failing test pass by removing the `expect_pass=fail` and the line on 125 currently